lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -657,31 +657,120 @@ def handle_user_input_data(model, climate_zone, default_hvac_building_type, defa
     user_airloops = @standards_data.key?('userdata_airloop_hvac') ? @standards_data['userdata_airloop_hvac'] : nil
-    # TODO: for now, it just work with economizer exceptions
     model.getAirLoopHVACs.each do |air_loop|
       air_loop_name = air_loop.name.get
       if user_airloops && user_airloops.length > 1
         user_airloops.each do |user_airloop|
           if air_loop_name == user_airloop['name']
             # gas phase air cleaning is system base - add proposed hvac system name to zones
-            if user_airloop.key?('economizer_exception_for_gas_phase_air_cleaning') &&
-               user_airloop['economizer_exception_for_gas_phase_air_cleaning'].downcase == 'yes'
-              air_loop.thermalZones.each do |thermal_zone|
-                thermal_zone.additionalProperties.setFeature('economizer_exception_for_gas_phase_air_cleaning', air_loop_name)
+            if user_airloop.key?('economizer_exception_for_gas_phase_air_cleaning') && !user_airloop['economizer_exception_for_gas_phase_air_cleaning'].nil?
+              if user_airloop['economizer_exception_for_gas_phase_air_cleaning'].downcase == 'yes'
+                air_loop.thermalZones.each do |thermal_zone|
+                  thermal_zone.additionalProperties.setFeature('economizer_exception_for_gas_phase_air_cleaning', air_loop_name)
+                end
               end
             end
             # Open refrigerated cases is zone based - add yes or no to zones
-            if user_airloop.key?('economizer_exception_for_open_refrigerated_cases') &&
-               user_airloop['economizer_exception_for_open_refrigerated_cases'].downcase == 'yes'
-              air_loop.thermalZones.each do |thermal_zone|
-                thermal_zone.additionalProperties.setFeature('economizer_exception_for_open_refrigerated_cases', 'yes')
+            if user_airloop.key?('economizer_exception_for_open_refrigerated_cases') && !user_airloop['economizer_exception_for_open_refrigerated_cases'].nil?
+              if user_airloop['economizer_exception_for_open_refrigerated_cases'].downcase == 'yes'
+                air_loop.thermalZones.each do |thermal_zone|
+                  thermal_zone.additionalProperties.setFeature('economizer_exception_for_open_refrigerated_cases', 'yes')
+                end
+              end
+            end
+            # Fan power credits
+            user_airloop.keys.each do |info_key|
+              if info_key.include?('fan_power_credit')
+                if !user_airloop[info_key].to_s.empty?
+                  if info_key.include?('has_')
+                    if user_airloop[info_key].downcase == 'yes'
+                      air_loop.thermalZones.each do |thermal_zone|
+                        if thermal_zone.additionalProperties.hasFeature(info_key)
+                          current_value = thermal_zone.additionalProperties.getFeatureAsDouble(info_key).to_f
```

Comment:
Is this here in case there are two air loops serving the same zone (e.g. DOAS and primary system)?

Reply:
Yes